apps/base-docs/tutorials/docs/4_smart-wallet-and-eoa-with-onchainkit.md

```diff
@@ -0,0 +1,275 @@
+
+---
+
+## Objectives
+
+By the end of this tutorial, you should be able to:
+
+- Create a connection experience using the `Wallet` component from [OnchainKit]
+- Customize the `Wallet` component to allow your users to connect from a chosen list of wallets
+- Give users the option to select the [Coinbase Smart Wallet] or EOA while connecting to your app with RainbowKit
+
+---
+
+## Prerequisites
+
+### Be familiar with modern, frontend web development
+
+In this tutorial, you'll be working with a React frontend built with [Next.js]. While you don't need to be an expert, we'll assume that you're comfortable with the basics.
+
+### Possess a general understanding of the EVM and smart contract development
+
+This tutorial assumes that you're reasonably comfortable writing basic smart contracts. If you're just getting started, jump over to our [Base Learn] guides and start learning!
+
+### Understand providers and connectors
+
+You'll need to be familiar with how to connect an onchain app to the blockchain with a provider. If you're not, start with the [Onchain App Development] section of [Base Learn], or at least complete the tutorial [Introduction to Providers].
+
+### Coinbase Wallets
+
+You need to have both the [Coinbase Wallet] and [Coinbase Smart Wallet] for this tutorial. You need to set up the [Coinbase Wallet] in advance, but you can create a smart wallet during the tutorial.
+
+---
+
+## The Default Experience
+
+Begin by using [An Onchain App in 100 Components] as a template to quickstart a new app. Click the green `Use this template` button, create a new repo, download it, install dependencies, and run the app.
+
+```bash
+bun install
+bun run dev
+```
+
+The demo app contains sections showing a number of [OnchainKit] components, including the [Wallet]. Click `Connect Wallet` in your browser with the Coinbase wallet active, and you'll get the expected experience for an EOA wallet user with the browser extension.
+
+![Default eoa](../../assets/images/smart-wallet/onchainkit-default-eoa.png)
+
+Next, open the app in a private browser window with extensions disabled, and try again. This time, you'll get the Smart Wallet experience. If you don't have one already, you can create one now. They're neat!
+
+![Default smart wallet](../../assets/images/smart-wallet/onchainkit-default-smart.png)
+
+Unfortunately, now that you've realized that the smart wallet is great, you've got a problem. Because you already have a Coinbase EOA wallet via the browser extension, you **can't** connect with your smart wallet **unless** you open a private window.
+
+## Customizing the List of Wallets
+
+Luckily, it isn't too hard to fix this with the [Wallet Aggregator].
+
+### Setting Up the Wallet Aggregator
+
+Open `src/app/page.tsx`. Here, you find the root page for the app. You can see the `<WalletComponents />`, which is what you used to log in and out.
```

Comment:
`<WalletComponents />` isn't present in Leo's latest version under `src/app/page.tsx`